### DISCUSSION TITLE
Are other Comfoair devices supported ?

### DISCUSSION THREAD

**Discussion by ScarlitoDoBrazil** [adorobis/hacomfoairmqtt#31] [Q&A] [answered]
Hi everyone.
I own a Comfoair 200 and I'm wondering if I could include it into Home Assistant.
I've seen a binding that seems pretty good for OpenHAB, and that is not limited to one device model : https://www.openhab.org/addons/bindings/comfoair/
As I'm using Home Assistant, I searched for a similar integration, and found yours. But it seems only available for Comfoair 350, and I don't know if it may work for another Zehnder device.
I also found the comfoconnect integration, but it's limited to "Q" devices if I'm understanding this right.

Any idea ?
Thanks for your help.

**Comment by adorobis** [comment, marked as answer]
According to the RS232 protocol documentation it is confirmed to work with the following models:
Zehnder ComfoAir 350 (not Q) (same as StorkAir 350)
Zehnder ComfoAir 550 (not Q) (same as StorkAir 550)
WHR930 from StorkAir
G90-380 from Wernig and Santos
370 DC from Paul

Not sure about the 200 model but if the openhab one works with it this one should work as well. "Q" models are not using RS232 so it won't work with this one.

**Reply by LutraVulgaris** [reply]
I own a comfoair e300 is that something I can manage? If yes can you get me going for starters?

**Reply by adorobis** [reply]
The E series is very limited in what you can connect it to. There are two versions: RF - with radio controllers and P with cable based switch (for selecting ventilation level). For the latter you could replace the switch with some smart relays which you could control from HA. For RF model you can only add a 0-10V input that you would control from HA. 
Neither of the versions allows for any communication module connection (neither RS232 nor the newer Zehnder modules) - so in either case your options are limited and this integration won't be useful.
I have only been able to find documentation in Dutch:
https://docplayer.nl/64709846-Comfoair-e-300-en-comfoair-e-400-handleiding-voor-de-servicemonteur-koeling-ventilatie-filtering.html

**Reply by LutraVulgaris** [reply]
Thabks, I'm dutch so thats not a problem. The housing is equipped with a domotica system (OneSmartControl) and that seems to be connected to the comfoair e300. Zehnder itself doesn't give the documentation for the protocol thats being used but they are reading something because it is connected via rs485/rs232.

**Reply by adorobis** [reply]
That's interesting, you would maybe need to open the connection box on comfoair and see what it is connected to. According to the E300 documentation there is no RS232/485 port on it. So maybe the OneSmartControl has its own module there? Or it is mybe just connected to the 0-10V input ?

**Reply by LutraVulgaris** [reply]
No i have a qubino 0-10v dimmer on that port. It is connected to the upper right port 
![20221104_105919](https://user-images.githubusercontent.com/110717536/209553583-57f78a98-27e8-4d13-adaa-bef4f8e1a945.jpg)

**Reply by adorobis** [reply]
Ok, to but how is the OneSmartControl connected to vicare?

**Reply by LutraVulgaris** [reply]
Vicare? Onesmart is connected to the growatt inverter, then it goes to the Zehnder comfoair e300 and then to the Mitsubishi Ecodan procon melcobems mini

**Reply by adorobis** [reply]
Sorry, I've meant comfoair

**Reply by adorobis** [reply]
So how is this inverter connected to comfoair? Did it have serial port? Or is there an additional module on it?

**Reply by LutraVulgaris** [reply]
The growatt also has a rs485 connection, it is all looped and connected to eachother. There is only one rs485 connection going to the onesmart but it reads all three. 

I can't tell what it reads from the comfoair e300 because the onesmart died a couple weeks back. However it is still connected.

Any idea if I could try something to get some readings? I got an ESP32 with a rs485 shield and maybe some yaml code?

**Reply by LutraVulgaris** [reply]
![20221104_105904](https://user-images.githubusercontent.com/110717536/209588438-156e8b9a-9f43-489d-ad24-9efea604b0b1.jpg)

**Reply by LutraVulgaris** [reply]
The blue block

**Reply by nkempe** [reply]
@LutraVulgaris replying in an old thread, hoping you can help me.
Have you managed to get the ComfoAir E300 connected to Home Assistant?

I have the same device, disappointed in the lack of connectivity of it, and I was wondering if the service ports would hold the key to connect it to Home Assistant. You seem to have some experience there. Can you tell me more?

**Comment by ScarlitoDoBrazil** [comment]
Thanks for the quick reply !
I'm in the process of recreating my Home Assistant installation, and I don't know when it will be done, but if I can test this anytime soon, I'll let you know how it behaves with my Comfoair 200.

Thanks !

**Comment by PieterRoelants** [comment]
Hey, here is my setup over the service port. Some sentences are in Dutch, but I’m sure you know how to use a translater 😉

Good luck!

Pieter

**Reply by nkempe** [reply]
@PieterRoelants  I think something went wrong in your comment, I don't see the setup you mention?
Also, Dutch is not an issue for me either. :)

**Comment by PieterRoelants** [comment]
I just replied this e-mail, with attachment. Please give me your e-mail address and I’ll resend it.

Pieter

**Reply by nkempe** [reply]
@PieterRoelants  alright, you can use github.sandy910@passmail.net

**Comment by LutraVulgaris** [comment]
If somebody has succeded in reading the Zehnder comfoair e300 I'd love to give it a shot also. My One Smart System could read all the sensors, but that system broke, but it used rs485 so there is some hope?
